### PR TITLE
[INLONG-682][Document] Sort out the format concept and related attributes

### LINKED ISSUE #682: [Feature][Doc] Sort out the format concept and related attributes
### Description

Sort out all the attributes of the format, which attributes are divided into categories, which are advanced attributes, and which are basic attributes.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by featzhang**
### Prepare a Pull Request

- [INLONG-682][Document] Add format related documents

- Fixes #682

### CODE REVIEW

**Review by yunqingmoswu — APPROVED** [review]

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+---
+title: Format
+sidebar_position: 7
+---
+
+## 什么是format?
```
Comment (healchow) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md:6]:
```suggestion
## 什么是 Format?
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+![](img/format_and_flink.png)
+
+如上图所示：
+
+Flink SQL在读写数据是，均采用Row的形式，其内部为Object数组`Object[]`, 数字中每个元素代表了一个Flink表的字段。
```
Comment (healchow) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md:12]: Please add one blank between the Chinese character and English character, thanks.

**Review by healchow — APPROVED** [review]
LGTM

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+sidebar_position: 7
+---
+
+## 什么是 Format?
+
+![](img/format_and_flink.png)
+
+如上图所示：
+
+Flink SQL 在读写数据是，均采用Row的形式，其内部为 Object 数组 `Object[]`，数字中每个元素代表了一个 Flink 表的字段。
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md:12]:
```suggestion
Flink SQL 在读写数据是，均采用 Row 的形式，其内部为 Object 数组 `Object[]`，数组中每个元素代表了一个 Flink 表的字段。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+字段的类型、名称、精度等信息，通过 `Schema` 来标示。
+
+Format 提供了两种接口：SerializationSchema 和 DeserializationSchema。
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md:15]:
```suggestion
Flink 的 Format 提供了两种接口：SerializationSchema 和 DeserializationSchema。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+- 当 Flink 往 MQ 写数据时，需要把 `Flink Row` 序列化为 `key-value` / `csv` / `Json` 等 Format,
+  调用了 `SerializationSchema#serialize` 方法，数据会序列化为 `Byte[]`，写入到 `MQ`。
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md:18]:
```suggestion
  这时调用了 `SerializationSchema#serialize` 方法，数据会序列化成 `Byte[]`，写入到 `MQ`。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+- 当 Flink 读取 MQ 的数据时，过程是相反的。从 MQ 读取数据，数据格式为 `byte[]`，反序列化为`Format`，再转换为 `Flink row`。
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md:19]:
```suggestion
- 当 Flink 读取 MQ 的数据时，该过程则相反：从 MQ 读取数据，数据格式为 `byte[]`，反序列化为`Format`，再转换为 `Flink row`。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+> 详情请查看代码 [`inlong-sort/sort-formats`](https://github.com/apache/inlong/tree/release-1.5.0/inlong-sort/sort-formats)
+
+## InLong 中的 Format
+
+![](img/the_format_in_inlong.png)
+
+InLong 作为一站式的数据集成平台，MQ（图中 Cache 部分）作为传输通道，同时实现 DataProxy 与 Sort 的解耦，扩展性更强。
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md:27]:
```suggestion
InLong 作为一站式的数据集成平台，将 MQ（图中 Cache 部分）作为传输通道，同时实现 DataProxy 与 Sort 的解耦，扩展性会更强：
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+1. DataProxy 上报数据时，需要将数据序列化成对应的格式（`SerializationSchema#serialize`）。
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md:29]:
```suggestion
- DataProxy 上报数据时，需要将数据序列化成对应的格式（`SerializationSchema#serialize`）。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+2. Sort 接收到数据，将 MQ 的数据反序列化（`DeserializationSchema#deserialize`）成 `Flink Row` ，通过 Flink SQL 写入到对应的存储。
```
Comment (dockerzhang) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/the_format_in_inlong.md:30]:
```suggestion
- Sort 接收到数据，将 MQ 的数据反序列化（`DeserializationSchema#deserialize`）成 `Flink Row` ，通过 Flink SQL 写入到对应的存储。
```

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**